src/EditorFeatures/Test2/InlineHints/CSharpInlineTypeHintsTests.vb

```diff
@@ -335,5 +335,30 @@ class A
 
             Await VerifyTypeHints(input)
         End Function
+
+        <WpfFact, Trait(Traits.Feature, Traits.Features.InlineHints)>
+        <WorkItem(49657, "https://github.com/dotnet/roslyn/issues/49657")>
+        Public Async Function TestWithTargetTypeNewInMethodArgument() As Task
+            Dim input =
+            <Workspace>
+                <Project Language="C#" CommonReferences="true">
+                    <Document>
+class A
+{
+    void M(string s) { }
+
+    void Main(string[] args) 
+    {
+        M(new{| string:|}())
+        {
+        }
+    }
+}
+                    </Document>
+                </Project>
+            </Workspace>
+
+            Await VerifyTypeHints(input)
```

Comment:
let's have a couple more cases.  For example, in a field intiializer, a local initialzer, and a parameter initializer (for example `void Foo(CancellationToken ct = new())`

Reply:
I added tests for FieldInitializer, LocalInitializer, ParameterInitializer, Return, and IfExpression in fec02db